### DISCUSSION TITLE
How to pass through params to SignIn() callback? I want to use invite codes to restrict sign-in?

### DISCUSSION THREAD

**Discussion by tombburnell** [nextauthjs/next-auth#4642] [Help]
### Question 💬

Hi,
I would like to provide people with invite codes that will allow them to sign in to the beta version of my site.

They will put the invite code and their email address into a form and attempt to Sign in (using email provider or facebook provider). 

During SignIn() callback I intend to check that the invite code is valid and not already used before allowing sign-in to continue.

I cannot find a way to retrieve either the callbackUrl or parameters passed in the 3rd argument during the signIn() callback. 

Is it possible? or is there an alternative method you recommend?

Thanks!

### How to reproduce ☕️

Example sign-In button..

<Button onClick={(e) => signIn("email", {
                    email:userEmail,
                    callbackUrl: "/?inviteCode=xyz",
                    inviteCode: "xyz"
                })}>Accept invitation</Button>
                
                
                
 async signIn({user, account, profile, email, credentials, ...rest}) {
            console.log("SignIn from ", user, account, profile, email, credentials, rest);

            //how to get callbackUrl or inviteCode??
            const inviteCode = ???

            return (checkInviteCode(inviteCode));            
 }
            

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
Additional params are passed through the third argument of `signIn`:

https://next-auth.js.org/getting-started/client#additional-parameters

**Reply by tombburnell** [reply]
But how to retrieve them in the signIn callback?

**Reply by tombburnell** [reply]
> Additional params are passed through the third argument of `signIn`:
> 
> https://next-auth.js.org/getting-started/client#additional-parameters

Hi - are you able to show me how I can access the parameters in the signin() callback?

**Reply by PaulNext** [reply]
I also encountered this problem, but on the official website https://next-auth.js.org/getting-started/client#additional-parameters I did not find out how to get twitter in [...nextauth].ts\ The third parameter passed by the front end after discord login is opposite, can you provide some examples? @balazsorban44 thanks

**Reply by PaulNext** [reply]
@tombburnell How does hi buddy solve this problem now? Is there an example to share?

**Reply by Abel1011** [reply]
Hello, you must use the advanced initialization https://next-auth.js.org/configuration/initialization#advanced-initialization to access the request and response objects. You will get the additional parameters in the body of the request.

**Comment by tombburnell** [comment]
Does anyone know how this can be done?
